evennia/commands/default/building.py

```diff
@@ -911,21 +911,29 @@ def func(self):
         """Implements the tunnel command"""
 
         if not self.args or not self.lhs:
-            string = "Usage: @tunnel[/switch] <direction> [= <roomname>" \
+            string = "Usage: @tunnel[/switch] <direction>[:typeclass] [= <roomname>" \
                      "[;alias;alias;...][:typeclass]]"
             self.caller.msg(string)
             return
-        if self.lhs not in self.directions:
+
+        exitshort = self.lhs.split(":")[0]
+
+        if exitshort not in self.directions:
             string = "@tunnel can only understand the following directions: %s." % ",".join(
                     sorted(self.directions.keys()))
             string += "\n(use @dig for more freedom)"
             self.caller.msg(string)
             return
+
         # retrieve all input and parse it
-        exitshort = self.lhs
         exitname, backshort = self.directions[exitshort]
         backname = self.directions[backshort][0]
 
+        if ":" in self.lhs:
+            exit_typeclass = ":" + self.lhs.split(":")[-1]
```

Comment:
You need to make sure to do `self.lhs.split(":", 1)[-1]` here or extra colons will confuse the parsing.